solution/parser/ecfr-parser/eregs/eregs.go

```diff
@@ -47,44 +50,102 @@ type ExistingPart struct {
 }
 
 // PostPart is the function that sends a part to the eRegs server
-func PostPart(ctx context.Context, p *Part) error {
+func PostPart(ctx context.Context, p *Part) (int, error) {
 	eregsPath, err := url.Parse(BaseURL)
 	if err != nil {
-		return err
+		return -1, err
 	}
-	return network.PostJSON(ctx, eregsPath, p, true, postAuth)
+	return network.SendJSON(ctx, eregsPath, p, true, postAuth, network.HTTPPost)
 }
 
 // PostSupplementalPart is the function that sends a supplemental part to eRegs server
-func PostSupplementalPart(ctx context.Context, p ecfr.Part) error {
+func PostSupplementalPart(ctx context.Context, p ecfr.Part) (int, error) {
 	eregsPath, err := url.Parse(BaseURL)
 	if err != nil {
-		return err
+		return -1, err
 	}
 	eregsPath.Path = path.Join(eregsPath.Path, "/supplemental_content")
-	return network.PostJSON(ctx, eregsPath, p, true, postAuth)
+	return network.SendJSON(ctx, eregsPath, p, true, postAuth, network.HTTPPost)
+}
+
+// GetTitle retrieves a title object from regcore in eRegs
+func GetTitle(ctx context.Context, title int) (*Title, int, error) {
+	emptyTitle := &Title{
+		Name: fmt.Sprintf("%d", title),
+		Contents: &ecfr.Structure{},
+		Exists: false,
+		Modified: false,
+	}
+
+	eregsPath, err := url.Parse(BaseURL)
+	if err != nil {
+		return emptyTitle, -1, err
+	}
+	// TODO: remove the following 2 lines on v3 move! BaseURL should point to v3.
+	if strings.HasSuffix(eregsPath.Path, "v2/") {
+		eregsPath.Path = eregsPath.Path[0:len(eregsPath.Path)-3] + "v3" // very bad!
```

Comment:
😂

Reply:
Yep lol this is bad. I toyed with having a V2BaseURL and a V3BaseURL simultaneously but honestly that was way more trouble than it's worth for some temporary code!